src/dbnode/storage/index/convert/convert_benchmark_test.go

```diff
@@ -0,0 +1,227 @@
+		tags: "dScMAAgAX19uYW1lX18DAGNwdQQAYXJjaAMAeDY0CgBkYXRhY2VudGVyCgB1cy13ZXN0LTJhCABob3N0bmFtZQYAaG9zdF82CwBtZWFzdXJlbWVudAoAdXNhZ2VfaWRsZQIAb3MLAFVidW50dTE2LjEwBAByYWNrAgAxMAYAcmVnaW9uCQB1cy13ZXN0LTIHAHNlcnZpY2UBADYTAHNlcnZpY2VfZW52aXJvbm1lbnQEAHRlc3QPAHNlcnZpY2VfdmVyc2lvbgEAMAQAdGVhbQMAQ0hJ",
+	},
+	{
+		id:   "e19fbmFtZV9fPSJuZ2lueCIsYXJjaD0ieDg2IixkYXRhY2VudGVyPSJ1cy1lYXN0LTFhIixob3N0bmFtZT0iaG9zdF80NCIsbWVhc3VyZW1lbnQ9ImhhbmRsZWQiLG9zPSJVYnVudHUxNi4wNExUUyIscmFjaz0iNjEiLHJlZ2lvbj0idXMtZWFzdC0xIixzZXJ2aWNlPSIyIixzZXJ2aWNlX2Vudmlyb25tZW50PSJzdGFnaW5nIixzZXJ2aWNlX3ZlcnNpb249IjEiLHRlYW09Ik5ZQyJ9",
+		tags: "dScMAAgAX19uYW1lX18FAG5naW54BABhcmNoAwB4ODYKAGRhdGFjZW50ZXIKAHVzLWVhc3QtMWEIAGhvc3RuYW1lBwBob3N0XzQ0CwBtZWFzdXJlbWVudAcAaGFuZGxlZAIAb3MOAFVidW50dTE2LjA0TFRTBAByYWNrAgA2MQYAcmVnaW9uCQB1cy1lYXN0LTEHAHNlcnZpY2UBADITAHNlcnZpY2VfZW52aXJvbm1lbnQHAHN0YWdpbmcPAHNlcnZpY2VfdmVyc2lvbgEAMQQAdGVhbQMATllD",
+	},
+	{
+		id:   "e19fbmFtZV9fPSJuZ2lueCIsYXJjaD0ieDg2IixkYXRhY2VudGVyPSJ1cy13ZXN0LTFhIixob3N0bmFtZT0iaG9zdF8yOSIsbWVhc3VyZW1lbnQ9IndhaXRpbmciLG9zPSJVYnVudHUxNS4xMCIscmFjaz0iMTUiLHJlZ2lvbj0idXMtd2VzdC0xIixzZXJ2aWNlPSI0IixzZXJ2aWNlX2Vudmlyb25tZW50PSJ0ZXN0IixzZXJ2aWNlX3ZlcnNpb249IjEiLHRlYW09Ik5ZQyJ9",
+		tags: "dScMAAgAX19uYW1lX18FAG5naW54BABhcmNoAwB4ODYKAGRhdGFjZW50ZXIKAHVzLXdlc3QtMWEIAGhvc3RuYW1lBwBob3N0XzI5CwBtZWFzdXJlbWVudAcAd2FpdGluZwIAb3MLAFVidW50dTE1LjEwBAByYWNrAgAxNQYAcmVnaW9uCQB1cy13ZXN0LTEHAHNlcnZpY2UBADQTAHNlcnZpY2VfZW52aXJvbm1lbnQEAHRlc3QPAHNlcnZpY2VfdmVyc2lvbgEAMQQAdGVhbQMATllD",
+	},
+	{
+		id:   "e19fbmFtZV9fPSJkaXNraW8iLGFyY2g9Ing2NCIsZGF0YWNlbnRlcj0iYXAtbm9ydGhlYXN0LTFjIixob3N0bmFtZT0iaG9zdF8zOCIsbWVhc3VyZW1lbnQ9IndyaXRlX3RpbWUiLG9zPSJVYnVudHUxNS4xMCIscmFjaz0iMjAiLHJlZ2lvbj0iYXAtbm9ydGhlYXN0LTEiLHNlcnZpY2U9IjAiLHNlcnZpY2VfZW52aXJvbm1lbnQ9InN0YWdpbmciLHNlcnZpY2VfdmVyc2lvbj0iMCIsdGVhbT0iU0YifQ==",
+		tags: "dScMAAgAX19uYW1lX18GAGRpc2tpbwQAYXJjaAMAeDY0CgBkYXRhY2VudGVyDwBhcC1ub3J0aGVhc3QtMWMIAGhvc3RuYW1lBwBob3N0XzM4CwBtZWFzdXJlbWVudAoAd3JpdGVfdGltZQIAb3MLAFVidW50dTE1LjEwBAByYWNrAgAyMAYAcmVnaW9uDgBhcC1ub3J0aGVhc3QtMQcAc2VydmljZQEAMBMAc2VydmljZV9lbnZpcm9ubWVudAcAc3RhZ2luZw8Ac2VydmljZV92ZXJzaW9uAQAwBAB0ZWFtAgBTRg==",
+	},
+}
+
+// BenchmarkFromSeriesIDAndTagIter-12    	  254090	      4689 ns/op
+func BenchmarkFromSeriesIDAndTagIter(b *testing.B) {
+	testData, err := prepareIDAndEncodedTags(b)
+	require.NoError(b, err)
+
+	decoderPool := serialize.NewTagDecoderPool(
+		serialize.NewTagDecoderOptions(serialize.TagDecoderOptionsConfig{}),
+		pool.NewObjectPoolOptions(),
+	)
+	decoderPool.Init()
+	tagDecoder := decoderPool.Get()
+	defer tagDecoder.Close()
+
+	b.ResetTimer()
+	for i := range testData {
+		tagDecoder.Reset(checked.NewBytes(testData[i].encodedTags, nil))
+
+		_, err := FromSeriesIDAndTagIter(testData[i].id, tagDecoder)
```

Comment:
This benchmarks not only the function that was optimized (`FromSeriesIDAndTagIter`), but also includes the overhead of decoding the encoded tags, which explains a smaller improvement for this function. Probably this is OK, but you might add a sentence explaining this to the PR comments where you provide benchmark results.